vtex/faststore
# Add: isAuthenticated for session

## What's the purpose of this pull request?

Possibility of adding validations in front-end components that depend on the user being logged in to generate different behaviors. 

## How it works?

When the user is logged the system will create the VtexIdclientAutCookie_account and when the client is logged out the VtexIdclientAutCookie_account is removed.
The session cookie have another behaviour so when the client is logged in or logged out the cookie will always exist and the client can be authenticated, identified or anonymous at the session.

Now we get the information of the namespace isAuthenticated from the session and we save this information at the IndexedDB to be available for the Front End if needed.

## How to test it?

- Create a session:

```
curl --location 'https://secure.vtexfaststore.com/api/sessions?sc=1' \
--header 'Content-Type: application/json' \
--data '{}
'
```

- Add the `storeUserEmail` to the session to identify the customer:

```
curl --location --request PATCH 'https://secure.vtexfaststore.com/api/sessions' \
--header 'Content-Type: application/json' \
--header 'Cookie: vtex_session={{vtex_session}}' \
--data-raw '{
    "public": {
        "storeUserEmail": {
            "value": "teste@faststore.com.br"
        }
    }
}'
```

You should expect this field to be saved at the IndexedDB:

<img width="719" alt="image" src="https://github.com/vtex/faststore/assets/67066494/5855194b-6f7b-4a31-b455-e776b6851005">
